src/code/z_en_a_keep.c

```diff
@@ -31,18 +31,18 @@ void EnAObj_Update1(ActorEnAObj* this, GlobalContext* globalCtx) {
     }
 }
 
-void EnAObj_Update2(ActorEnAObj* this, GlobalContext* globalCtx) {
+void EnAObj_Update2(EnAObj* this, GlobalContext* globalCtx) {
     if (func_800B867C((Actor*)this, globalCtx) != 0) {
-        this->update = (ActorFunc)EnAObj_Update1;
+        this->actionFunc = (EnAObjActionFunc)EnAObj_Update1;
     }
 }
 
-void EnAObj_Update(ActorEnAObj* this, GlobalContext* globalCtx) {
-    (this->update)((Actor*)this, (GlobalContext*)globalCtx);
+void EnAObj_Update(EnAObj* this, GlobalContext* globalCtx) {
+    (this->actionFunc)((struct EnAObj*)this, (GlobalContext*)globalCtx);
     Actor_SetHeight((Actor*)this, 45.0f);
     CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, (Collider*)&this->collision);
 }
 
-void EnAObj_Draw(ActorEnAObj* this, GlobalContext* globalCtx) {
+void EnAObj_Draw(EnAObj* this, GlobalContext* globalCtx) {
     func_800BDFC0(globalCtx, enAObjDisplayLists[this->base.params]);
 }
```

Comment:
`thisx`, and recast if room.

Reply:
Regalloc gets messed up if i recast here, so just using the thisx